### ISSUE TITLE
Problem with nullability inference

### ISSUE THREAD

**Issue by nnemkin** [microsoft/pyright#1198]
**Describe the bug**
The following code produces an unexpected error:

```python
from __future__ import annotations

def f(key: str, msg: str | None = None) -> str:
    if msg is None:
        msg = {"a": "b"}.get(key, "c")  # right hand side is "str", but msg stays "str | None" after assignment
    return msg  # ERROR: Expression of type "str | None" cannot be assigned to return type "str"
```

**Expected behavior**
At the point of return, `msg` should be inferred to be of type `str`.

I believe it's a bug and not just lack of inference power, becase inserting a temporary solves the problem:
```python
def f(key: str, msg: str | None = None) -> str:
    if msg is None:
        tmp = {"a": "b"}.get(key, "c")  # tmp is "str"
        msg = tmp  # msg is "str"
    return msg  # No error
```

**VS Code extension or command-line**
Pylance v2020.11.2

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next version.

**Comment by erictraut** [comment]
This is now addressed in Pyright 1.1.88, which I just published. It will also be included in the next release of Pylance.
